Assets/Features/Building/Scripts/Demos/CursorSelectorDemo.cs

```diff
@@ -8,18 +9,32 @@ namespace Features.Building.Scripts.Demos
 {
     public class CursorSelectorDemo : MonoBehaviour
     {
-        [SerializeField] private List<BuildableObject> _buildable;
         [SerializeField] private Cursor _cursor;
+        [SerializeField] private TMP_Dropdown _dropdown;
 
         private void Start()
         {
-            GetComponent<TMP_Dropdown>().onValueChanged.AddListener(Select);
+            UpdateDropDown();
+
+            _dropdown.onValueChanged.AddListener(Select);
+            _dropdown.RefreshShownValue();
+            
+            GameManager.Instance.EventManager.Subscribe(EventId.UnlockBuilding, (args) => UpdateDropDown());
         }
 
-        private void Select(int index)
+        private void UpdateDropDown()
         {
-            _cursor.ChangeSelectedBuildable(_buildable[index]);
+            _dropdown.options.Clear();
+            
+            foreach (BuildingStatus buildingStatus in GameManager.Instance.BuildingManager.BuildingStatuses.Where(x => x.IsUnlocked))
+            {
+                _dropdown.options.Add(new TMP_Dropdown.OptionData(buildingStatus.BuildableObject.name, buildingStatus.BuildableObject.BuildItems[0].Tile.sprite));
+            }
         }
 
+        private void Select(int index)
+        {
+            GameManager.Instance.BuildingManager.ChangeSelectedBuildable(GameManager.Instance.BuildingManager.BuildingStatuses[index].BuildableObject);
```

Comment:
This implementation seems way to difficult. Using `ChangeSelectedBuildable(index)` or something might be more straightforward for the user. Also improves readability

Reply:
Was inderdaad stupid